src/v2/Components/FlashBanner/__tests__/FlashBanner.jest.tsx

```diff
@@ -1,32 +1,181 @@
-import { mount } from "enzyme"
 import "jest-styled-components"
 import React from "react"
-import { FlashBanner } from "v2/Components/FlashBanner"
+import { BaseFlashBanner } from "v2/Components/FlashBanner"
+import { graphql } from "react-relay"
+import { Banner } from "@artsy/palette"
+import { flushPromiseQueue, renderRelayTree } from "v2/DevTools"
+import { SystemContextProvider } from "v2/Artsy"
+
+jest.unmock("react-tracking")
+jest.unmock("react-relay")
 
 const { location: originalLocation } = window
 
-describe("FlashBanner", () => {
-  afterEach(() => {
-    window.location = originalLocation
+const getWrapper = async ({
+  data,
+  mutationResults = {},
+  queryString: search = "",
+  props: passedProps = {},
+}) => {
+  delete window.location
+  window.location = { search } as any
+
+  return await renderRelayTree({
+    Component: props => {
+      return (
+        <SystemContextProvider>
+          <BaseFlashBanner {...props} {...passedProps} />
+        </SystemContextProvider>
+      )
+    },
+    query: graphql`
+      query FlashBannerTestQuery @raw_response_type {
+        me {
+          canRequestEmailConfirmation
+        }
+      }
+    `,
+    variables: {},
+    mockData: data,
+    mockMutationResults: mutationResults,
   })
+}
+
+afterEach(() => {
+  window.location = originalLocation
+})
+
+describe("FlashBanner", () => {
+  it("renders nothing if no banner applies", async () => {
+    const wrapper = await getWrapper({
+      data: {
+        me: null,
+      },
+    })
 
-  it("renders a banner based on a messageCode prop", () => {
-    const wrapper = mount(<FlashBanner messageCode="confirmed" />)
-    expect(wrapper.text()).toContain("Your email has been confirmed.")
+    expect(wrapper.find(Banner).exists()).toBeFalsy()
   })
 
-  it("returns nothing without complaint if the message code is not supported", () => {
-    const wrapper = mount(<FlashBanner messageCode="bad_message" />)
-    expect(wrapper.html()).toBeNull()
+  it("renders based on props.me.canRequestEmailConfirmation from relay", async () => {
```

Comment:
this test might not be needed now considering the email confirmation CTA describe below